source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -88,13 +246,22 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Authenticator::Callback
   callback_ = callback;
 
   ENVOY_LOG(debug, "Jwt authentication starts");
+
+  ASSERT(headers.Path());
+  const auto& ret = tryMatch(headers.Path()->value());
+  if (!ret.first || ret.second->getRequirementVerifier() == nullptr) {
```

Comment:
Current approach will be very difficult to extend to support multiple tokens.  Especially with such requirements as 
1) require A and B,   or 2) require A or B.

Ideally,  for 1), if A fail,  not need to verify B.   for 2)  if A success, not need to verify B

But single token is the most common use case.   We can worry about supporting multiple tokens later.
Please add // TODO